clients/ts/signalr/src/Transports.ts

```diff
@@ -313,17 +314,17 @@ export class LongPollingTransport implements ITransport {
     public onclose: TransportClosed;
 }
 
-function getDataLength(data: any): number {
-    let length: number = null;
+function getDataDetail(data: any): string {
+    let length: string = null;
     if (data instanceof ArrayBuffer) {
-        length = data.byteLength;
-    } else if (data instanceof String) {
-        length = data.length;
+        length = `Binary data of length ${data.byteLength}`;
+    } else if (typeof data === "string") {
```

Comment:
Whats wrong with `instanceof String`?

Reply:
`instanceof String` will match `new String("Hello World")` 🤷‍♂️ 

https://stackoverflow.com/questions/16792051/how-to-instanceof-a-primitive-string-string-literal-in-javascript

Reply:
Yeah, `instanceof String` **only** matches `new String(...)`, not literals. So, `typeof` is the idomatic way to do this here

Reply:
![image](https://user-images.githubusercontent.com/7574/38122239-a57e3902-3388-11e8-934d-0efb2b7f45a4.png)

Reply:
JavaScript is weird yo.